### ISSUE TITLE
Restricting the user and space pickers for privacy

### ISSUE THREAD

**Issue by Ace2213** [humhub/humhub#3679]
I’m trying to restrict the names that show up when @mentioning a user or choosing a message recipient.

I found the $userRole directive in [Userpicker.php](https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/widgets/UserPicker.php) but changing it to “friends” or “friendsonly” or “friends-only” doesn’t work

Ideally, the user picker should only show friends (if friendship is enabled). Within spaces, it should additionally show space members.

The space picker should only show public spaces and spaces that you’re a member of. 

I understand this isn't meaningful for smaller networks, but I think any network that enables friendship should have this behavior by default. And the space picker should never list private spaces in any scenario.

**Comment by buddh4** [comment]
Yes, a better picker pre selection would be great, also for example auto suggest users which are part of the current discussion (as here in github)
